File: Client/src/app/shared/models/tournament.ts
export interface CreateTeamRequest {
  name: string;
  manager: string;
  numberOfPlayers: number;
  logoUrl?: string;
  tournamentId: string;
}

export interface UpdateTeamRequest {
  name: string;
  manager: string;
  numberOfPlayers: number;
  logoUrl?: string;
}

export interface TeamSearchRequest {
  searchTerm?: string;
  tournamentId?: string;
  pageNumber?: number;
  pageSize?: number;
}

export interface Team {
  id: string;
  name: string;
  manager: string;
  numberOfPlayers: number;
  logoUrl?: string;
  tournamentId: string;
  tournamentName: string;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateTournamentRequest {
  name: string;
  description: string;
  location?: string;
  registrationDeadline?: string;
  startDate: string;
  endDate: string;
  maxTeams: number;
  entryFee?: number;
  prize?: number;
  contactEmail?: string;
  bannerUrl?: string;
}

export interface UpdateTournamentRequest {
  name: string;
  description: string;
  location?: string;
  registrationDeadline?: string;
  startDate: string;
  endDate: string;
  maxTeams: number;
  entryFee?: number;
  prize?: number;
  contactEmail?: string;
  bannerUrl?: string;
  status?: string;
}

export interface TournamentSearchRequest {
  searchTerm?: string;
  status?: string;
  location?: string;
  startDateFrom?: string;
  startDateTo?: string;
  pageNumber?: number;
  pageSize?: number;
}

export interface Tournament {
  id: string;
  name: string;
  description: string;
  location?: string;
  registrationDeadline?: string;
  startDate: string;
  endDate: string;
  maxTeams: number;
  entryFee?: number;
  prize?: number;
  contactEmail?: string;
  bannerUrl?: string;
  status: string;
  organizer: string;
  organizerName: string;
  createdAt: string;
  teamCount: number;
}

export interface TeamFormData {
  name: string;
  city: string;
  coach: string;
  players: number;
  logoUrl?: string;
}

export interface TournamentFormData {
  name: string;
  description: string;
  startDate: string;
  endDate: string;
  maxTeams: number;
  location?: string;
  registrationDeadline?: string;
  entryFee?: number;
  prizePool?: number;
  contactEmail?: string;
}

File: Server/PhantomGG.Repository/Entities/Tournament.cs
using System;
using System.Collections.Generic;

namespace PhantomGG.Repository.Entities;

public partial class Tournament
{
    public Guid Id { get; set; }

    public string Name { get; set; } = null!;

    public string Description { get; set; } = null!;

    public string Location { get; set; } = null!;

    public string? BannerUrl { get; set; }

    public string? LogoUrl { get; set; }

    public DateTime RegistrationStartDate { get; set; }

    public DateTime RegistrationDeadline { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime? EndDate { get; set; }

    public int MinTeams { get; set; }

    public int MaxTeams { get; set; }

    public string Status { get; set; } = null!;

    public Guid OrganizerId { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public bool IsPublic { get; set; }

    public virtual ICollection<Match> Matches { get; set; } = new List<Match>();

    public virtual User Organizer { get; set; } = null!;

    public virtual ICollection<TournamentTeam> TournamentTeams { get; set; } = new List<TournamentTeam>();
}
